include/rect.h

```diff
@@ -0,0 +1,67 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2023-2023  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_RECTANGLE_H
+#define DOSBOX_RECTANGLE_H
+
+#include "math_utils.h"
+
+namespace DosBox {
+
+// Struct to represent rectangles.
+struct Rect {
+	constexpr Rect() = default;
+
+	constexpr Rect(const float _w, const float _h)
+	        : x(0.0f),
+	          y(0.0f),
+	          w(_w),
+	          h(_h)
+	{}
+
+	constexpr Rect(const int _w, const int _h)
+	        : x(static_cast<float>(0.0f)),
+	          y(static_cast<float>(0.0f)),
+	          w(static_cast<float>(_w)),
+	          h(static_cast<float>(_h))
+	{}
+	constexpr Rect(const float _x, const float _y, const float _w, const float _h)
+	        : x(_x),
+	          y(_y),
+	          w(_w),
+	          h(_h)
+	{}
+
+	constexpr Rect(const int _x, const int _y, const int _w, const int _h)
```

Comment:
For the argument names, just a minor suggestion to use verbose (or expanded) form because these are usually expanded by IDEs, so the expanded form is a bit more self-documenting for users of the class.

(for example: `x` and `y` often serve as width and height in lots of places, so spelling out the args as: `x_pos`, `y_pos`, `width`, `height` will help differentiate them)

This would then also avoids the need for all the underscore prefixes.

(probably unwarranted, but I try to always stay away of underscore prefixes just to avoid  inmplementation-specific reservations and name conflicts):

---

2003 C++ Standard:

    17.4.3.1.2 Global names [lib.global.names]

        Each name that begins with an underscore is reserved to the implementation for use as a name in the global namespace.165

    7.1.3 Reserved identifiers

        All identifiers that begin with an underscore and either an uppercase letter or another underscore are always reserved for any use.

https://stackoverflow.com/a/228797

Reply:
Good suggestion @kcgen; I wasn't in love with the underscores either.

Reply:
Right on, @johnnovak!